### PR TITLE
[ME-7]added a retry feature to some of the API calls

### PR DESCRIPTION
**Pull request by Azbagheri**
… requests

### CODE REVIEW

File: Sources/Trigger/BitriseClient.swift [hunk 1]
```diff
@@ -1,239 +1,230 @@
...
+        case 0:  return "Build TIMED OUT base on mobile trigger internal setting"
+        case 1:  return "Build finished, with success"
+        case 2:  return "Build finished, with error"
+        case 3:  return "Build was aborted"
+        default: return "Build status did not match any of the values 0, 1, 2 or 3.\n This is very strange!"
+        }
     }
+    
+    private func handleRequest(request: URLRequest) -> Data? {
+        var requestHandled = false
```
Comment (aroswell) [inline Sources/Trigger/BitriseClient.swift]: Try using break keyword instead of having this var.

File: Sources/Trigger/BitriseClient.swift [hunk 2]
```diff
@@ -1,239 +1,230 @@
...
+            guard let res = response as? HTTPURLResponse else {
+                print("HTTPURLResponse returned nil")
+                exit(1)
+            }
+            let returnCode = res.statusCode
+            
+            if returnCode == 503 {
+                print("Response code for checkBuildStatus was 503, retrying. Attempt", retry)
+                sleep(pollingInterval)
+            } else if returnCode == 200 {
```
Comment (aroswell) [inline Sources/Trigger/BitriseClient.swift]: Just consider 503 or not. So you would just need and `else` statement here.

File: Sources/MobileTrigger/main.swift [hunk 3]
```diff
@@ -68,99 +68,99 @@ var bitriseClient = BitriseClient(projectConfig: projectConfig)
...
-    }
-    let currentBuildStatusText = res.data.statusText
-    if previousBuildStatusText != currentBuildStatusText {
-        print("Build", res.data.statusText)
-        previousBuildStatusText = currentBuildStatusText
+    while buildStatus == 0 {
+        sleep(5)
+        guard let res = bitriseClient.checkBuildStatus(slug: buildSlug) else {
+            print(":!ERROR - checkBuildStatus returned nil")
+            continue
```
Comment (aroswell) [inline Sources/MobileTrigger/main.swift]: Why are you continuing here?

File: Sources/Trigger/BitriseClient.swift (hunk 1 above)
Comment (Azbagheri) [reply Sources/Trigger/BitriseClient.swift]: Done!

File: Sources/Trigger/BitriseClient.swift (hunk 2 above)
Comment (Azbagheri) [reply Sources/Trigger/BitriseClient.swift]: Done!

File: Sources/MobileTrigger/main.swift (hunk 3 above)
Comment (Azbagheri) [reply Sources/MobileTrigger/main.swift]: fixed it!

**Pull request merged**